service/models.py

```diff
@@ -51,6 +51,15 @@ def create(self):
         db.session.add(self)
         db.session.commit()
 
+    def update(self):
+        """
+        update a recommendation type to the database
+        """
+        if not self.relationship:
```

Comment:
To test this line, have a test case say test_update_empty_relationship() and do not set relationship in your input. Then DataValidationError should be expected.

Reply:
Could you pls help look at test_update_recommendation_no_relation_type(self)
I try to assign null to relation type in json file, but it raises error when deserialize it. A little confused how to test this one

 service/routes.py line 92 in update_recommendations
      recommendation.deserialize(request.get_json())
    service/models.py line 83 in deserialize
      raise DataValidationError(
   DataValidationError: Invalid YourResourceModel: body of request contained bad or no data
   -------------------- >> begin captured logging << --------------------
   root: DEBUG: <enum 'Type'>
   root: DEBUG: {'product_id1': 1, 'product_id2': 7, 'relationship': 'CROSS_SELL'}
   root: DEBUG: {"product_id1": 1, "product_id2": 7, "relationship": " "}
   --------------------- >> end captured logging << ---------------------